docs/dev/howtoTestRepository.md

```diff
@@ -0,0 +1,156 @@
+
+#### @BeforeClass 
+To prepare test environments inside the Test Class the `@BeforeClass` annotation can be used. it will cause to run before any Testmethod is called. So you can for exmaple mock some simple Test data for all Tests.
+
+#### @After
+To Clean up prepared runtime variables or other Data you can call an `@After` Method it runs each time a test method has been finished.
+
+#### @AfterClass 
+To Clean up prepared runtime variables or other Data you can call an `@AfterClass` Method it runs after all test method has been finished.
+
+#### @Ignore
+Will tell junit to ignore, a test. Maybe its not working anymore or its just deprecated.
+
+
+
+## Add Test Methods
+For basic testing in the repository, we created two Testcases for each type.
+```
+  @Test
+  public void getInstanceXml() throws Exception {
+  // Getting Xml from a API, checking if its equal to the expected local file.
+  }
+  @Test
+  public void getServicetemplate() throws Exception {
+  // Getting JSON from a API, checking if its equal to the expected local file.
+  }
+```
+
+These are the two basic methods to test the capability of the API, to serve JSON and XML-Filetypes as response.
+
+## Things to prepare on Test
+Before running the Basic Test logic like in the Chapter [Add Test Methods](##Add Test Methods)
+you have to get a Github commit-id from the [Test-Repository](https://github.com/winery/test-repository).
+
+In the following example is shown how to get a revision:
+```
+  @Test
+  public void getServicetemplate() throws Exception {
+    this.setRevisionTo("c25aa724201824fce6eddcc7c35a666c6e015880");
+    // Test Logic 
+  }
+```
+## Implement a Testing Logic
+
+To Test the Logic you can simply implement this two liner Test, it will check out the Revision `c25aa724201824fce6eddcc7c35a666c6e015880` from Github Repository located under [Test-Repository](https://github.com/winery/test-repository).
+After wards it will send a `GET`Request against the Backend initialized by the GIT-Repository Data. The Response will be tested for equality against a File with the expected Result.
+  
+If the Result is equal the Test will become marked as successful.  
+```
+	@Test
+ 	public void getInstanceXml() throws Exception {
+ 		this.setRevisionTo("c25aa724201824fce6eddcc7c35a666c6e015880");
+ 		this.assertGet(INSTANCE_URL, INSTANCE_LOCAL_PATH);
+ 	}
+```
+## Run a Test or Test Class
+If you want to run a Test you can simply Run it from Menu in your Eclipse or IntelliJ IDE.
+
+## Appendix 
+
+Here we have a simple implementation of the `NodeTypeImplementationResourceTest`. 
+````
+
+package org.eclipse.winery.repository.resources.entitytypeimplementations.nodetypeimplementations;
+
+import org.eclipse.winery.repository.resources.AbstractResourceTest;
+
+import org.junit.Test;
+
+public class NodeTypeImplementationResourceTest extends AbstractResourceTest {
+	private final String ENTITY_TYPE = "nodetypeimplementations/";
```

Comment:
Please do not use constants. On that level, it is not optimal, because
a) it is not generalized using a super class
b) it is not consistent to the other tests